File: software/examples/csharp/ExampleCallback.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your CO2 Bricklet

	// Callback function for CO2 concentration callback (parameter has unit ppm)
	static void CO2ConcentrationCB(BrickletCO2 sender, int co2Concentration)
	{
		Console.WriteLine("CO2 Concentration: " + co2Concentration + " ppm");
	}

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletCO2 co2 = new BrickletCO2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register CO2 concentration callback to function CO2ConcentrationCB
		co2.CO2ConcentrationCallback += CO2ConcentrationCB;

		// Set period for CO2 concentration callback to 1s (1000ms)
		// Note: The CO2 concentration callback is only called every second
		//       if the CO2 concentration has changed since the last call!
		co2.SetCO2ConcentrationCallbackPeriod(1000);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var co2 = new Tinkerforge.BrickletCO2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Set Period for co2 concentration callback to 1s (1000ms)
        // Note: The co2 concentration callback is only called every second if the
        // co2 concentration has changed since the last call!
        co2.setCO2ConcentrationCallbackPeriod(1000);
    }
);

// Register position callback
co2.on(Tinkerforge.BrickletCO2.CALLBACK_CO2_CONCENTRATION,
    // Callback function for co2 concentration callback (parameter has unit ppm)
    function(co2Concentration) {
        console.log('CO2 Concentration: '+co2Concentration+' ppm');
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
